File: Tree/Binary Tree.cpp
#include <iostream>
using namespace std;

class Node
{
public:
    int data;
    Node *left;
    Node *right;

    Node(int value)
    {
        data = value;
    }
};

void preorder(Node *keep)
{
    if (keep != NULL)
    {
        cout << keep->data << " ";
        preorder(keep->left);
        preorder(keep->right);
    }
}

void postorder(Node *keep)
{
    if (keep != NULL)
    {
        postorder(keep->left);
        postorder(keep->right);
        cout << keep->data << " ";
    }
}

void inorder(Node *keep)
{
    if (keep != NULL)
    {
        inorder(keep->left);
        cout << keep->data << " ";
        inorder(keep->right);
    }
}

int main()
{

    //Creating nodes but they aren't interconnected
    Node *root = new Node(1);
    Node *first = new Node(2);
    Node *second = new Node(3);
    Node *third = new Node(4);
    Node *fourth = new Node(5);
    Node *fifth = new Node(6);
    Node *sixth = new Node(7);

    // Connecting nodes
    root->left = first;
    root->right = second;
    first->left = third;
    first->right = fourth;
    second->left = fifth;
    second->right = sixth;

    cout << "Pre-order Traversal: ";
    preorder(root);

    cout << endl
         << "In-order Traversal : ";
    inorder(root);

    cout << endl
         << "Post-order Traversal : ";
    postorder(root);

    return 0;
}
// Pre - order Traversal : 1 2 4 5 3 6 7 
// In - order Traversal : 4 2 5 1 6 3 7 
// Post - order Traversal : 4 5 2 6 7 3 1

File: Tree/Binary Tree.py
class Node:
    def __init__(self,value):
        self.data=value
        self.left=None
        self.right=None

def preorderTraversal(keep):
    if (keep != None):
        print(keep.data,end=" ")
        preorderTraversal(keep.left)
        preorderTraversal(keep.right)

def inorderTraversal(keep):
    if (keep != None):
        inorderTraversal(keep.left)
        print(keep.data,end=" ")
        inorderTraversal(keep.right)

def postorderTraversal(keep):
    if (keep != None):
        postorderTraversal(keep.left)
        postorderTraversal(keep.right)
        print(keep.data,end=" ")

#creating 3 Nodes in heap memory
root =Node(1)
first =Node(2)
second =Node(3)
third =Node(4)
fourth =Node(5)
fifth =Node(6)
sixth =Node(7)

#connecting Nodes
root.left=first
root.right=second
first.left=third
first.right=fourth
second.left=fifth
second.right=sixth

print("Preorder Traversal : ",end=' ')
preorderTraversal(root)

print("\n Inorder Traversal : ",end=' ')
inorderTraversal(root)

print("\n Postorder Traversal : ",end=' ')
postorderTraversal(root)

